systemd/systemd.go

```diff
@@ -205,6 +209,13 @@ func New(rootDir string, mode InstanceMode, rep reporter) Systemd {
 	return &systemd{rootDir: rootDir, mode: mode, reporter: rep}
 }
 
+// NewEmulationMode returns a Systemd that runs in emulation mode where
+// systemd is not really called, but instead its functions are emulated
+// by other means.
+func NewEmulationMode(rootDir string, rep reporter) Systemd {
+	return &systemd{rootDir: rootDir, mode: SystemMode, emulation: true, reporter: rep}
```

Comment:
More of a general question, but maybe we should provide a separate implementation for prebaking mode? AIUI we still won't be able to or not want to  execute a number of methods of the Systemd interface.

Reply:
it actually sounds like a very good idea to me, considering that  in emulation we don't call any systemd at all, no?

Reply:
```
type Systemd interface {
	DaemonReload() error
	Enable(service string) error
	Disable(service string) error
	Start(service ...string) error
	StartNoBlock(service ...string) error
	Stop(service string, timeout time.Duration) error
	Kill(service, signal, who string) error
	Restart(service string, timeout time.Duration) error
	Status(units ...string) ([]*UnitStatus, error)
	IsEnabled(service string) (bool, error)
	IsActive(service string) (bool, error)
	LogReader(services []string, n int, follow bool) (io.ReadCloser, error)
	AddMountUnitFile(name, revision, what, where, fstype string) (string, error)
	RemoveMountUnitFile(baseDir string) error
	Mask(service string) error
	Unmask(service string) error
}
```
The interface is quite large, but I suppose most of these will be nops or returning `errors.New("not implemented")`

Reply:
> it actually sounds like a very good idea to me, considering that in emulation we don't call any systemd at all, no?

Indeed, no systemd calls.

Yep, I like that idea too, thanks @bboozzoo .